### ISSUE TITLE
Store weather history

### ISSUE THREAD

**Issue by akosel** [akosel/freyr#1]
Before triggering a sprinkler run, weather data is used. Specifically it is currently used to:

- fetch precipitation from rain in the previous 3 days to determine if the sprinkler needs to run at all
- fetch the high temperature for the day and apply a multiplier if it is too hot (x1.3) or too cold (x0.7)

There is a chance the weather API could go down temporarily. Or more likely that the Pi itself will lose its network connection. In this case we can choose to error out until we reconnect (the current behavior) or try to use values pulled from previously-stored weather API queries. 

To summarize the work needed:

- create a new WeatherHistory model
- create a systemd timer job that queries for weather data every hour
- update schedule code to use weather history in case of error. 

Additionally this lets us replace the current in-memory caching behavior on the weather API responses.
